### ISSUE TITLE
save model

### ISSUE THREAD

**Issue by asdasdfasdfsd** [dumyy/handpose#4]
Hello. 
Thank you for sharing your excellent project.
I have a question while analyzing your code.
According to your code, it seems to calculate the error for the test set for each epoch and store the model with the smallest error. 
Is this usual way to use this model for testing?
Thank you

**Comment by dumyy** [comment]
You had better choose a validation set wheh training, and choose to save the model according to the 
validation results. Here just is a lazy way to save the model.   It would do nothing if you don't know the test label in this lazy method ,eg  training in HANDS 17 dataset.

**Comment by asdasdfasdfsd** [comment]
Thanks for reply.
Yes you are right, but according to your shared code, it seems that you used test set as a validation set for msra, icvl, and nyu databases. 
Is is right what i understood?

**Comment by dumyy** [comment]
you can think so.
